Runtime/Scripts/OscSocket.cs

```diff
@@ -74,15 +106,26 @@ void Serve()
                     break;
                 }
             }
-            
+
+            m_ThreadWakeup.Dispose();
             Profiler.EndThreadProfiling();
         }
 
         public void Dispose()
         {
             if (m_Disposed) return;
-            m_Socket.Close();
-            m_Socket.Dispose();
+            if (m_ThreadWakeup != null)
+            {
+                // thread running, let it dispose itself async
+                m_CloseRequested = true;
```

Comment:
Do you think it's possible for two threads to call this at the same time?

Reply:
In practice, no, but even if, this method is thread-safe, because `AutoResetEvent.Set` can be called multiple times and `m_CloseRequested = true` is a set-without-compare.